### PR TITLE
Add loaders prop to Layer class

### PR DESCRIPTION
**Pull request by Pessimistress**
For #5263

#### Change List
- Add `loaders` prop to `Layer` class
- `MVTLayer` and `TerrainLayer` support overriding of the default loader
- `ScenegraphLayer` has GLTFLoader built-in
- `Tile3DLayer` supports `loaders` in addition to `loader`
- Documentation
- Upgrade guide

### CODE REVIEW

File: docs/upgrade-guide.md [hunk 1]
```diff
@@ -22,9 +22,10 @@ new Deck({
 
 A legacy field `pickingInfo.lngLat` has been removed. Use `pickingInfo.coordinate` instead.
 
-### MVTLayer
+### Layers
 
-- `onHover` and `onClick` callbacks now throw the `info.object` feature coordinates in WGS84 standard.
+- `MVTLayer`'s `onHover` and `onClick` callbacks now yield the `info.object` feature coordinates in WGS84 standard.
+- `ScenegraphLayer` now have built-in support for `GLTFLoader`. It's no longer necessary to call `registerLoaders` before using it.
```
Comment (kylebarron) [inline docs/upgrade-guide.md:28]:
```suggestion
- `ScenegraphLayer` now has built-in support for `GLTFLoader`. It's no longer necessary to call `registerLoaders` before using it.
```

**Review by kylebarron — APPROVED** [review]

File: modules/geo-layers/src/tile-3d-layer/tile-3d-layer.js [hunk 2]
```diff
@@ -67,7 +67,14 @@ export default class Tile3DLayer extends CompositeLayer {
   }
 
   async _loadTileset(tilesetUrl) {
-    const {loader, loadOptions} = this.props;
+    const {loadOptions} = this.props;
+
+    // TODO: deprecate `loader` in v9.0
```
Comment (juneidysoo) [inline modules/geo-layers/src/tile-3d-layer/tile-3d-layer.js:72]: Hey guys, just out of curiosity, is there any link to documentation or discussion on why loader will be deprecated in v9?

I was just looking around at how the loaders are implemented and surprised to see the comment.

Comment (Pessimistress) [reply modules/geo-layers/src/tile-3d-layer/tile-3d-layer.js:72]: `loader` existed on `Tile3DLayer` prior to `loaders` being supported by all layers. We are not only going to use `loaders` moving forward for consistency.

Comment (juneidysoo) [reply modules/geo-layers/src/tile-3d-layer/tile-3d-layer.js:72]: Fantastic! Thanks for the response and clarification!

**Pull request merged**